app/scripts/background.js

```diff
@@ -291,20 +316,19 @@ function setupController(initState, initLangCode) {
     }
   }
 
+  if (process.env.ENABLE_MV3 && remoteSourcePort) {
+    connectRemote(remoteSourcePort);
```

Comment:
same question here. Why not simply add the listener (`browser.runtime.onConnect.addListener(connectRemote);`) at the root of the file?

Reply:
`connectRemote` use controller object which is initialised in method containing it [here](https://github.com/MetaMask/metamask-extension/pull/14047/files/146e7efdec2f18c24441fbe2d77d44b02ddcdbf9#diff-215d2cf6fb64ee62b9f484e7bbdd33246739e39b854e3cc8c2245b026352c12aR240).